backend/src/org/commcare/xml/DetailParser.java

```diff
@@ -148,10 +148,6 @@ public Detail parse() throws InvalidStructureException, IOException, XmlPullPars
                     parser.nextTag();
                     DetailTemplate template;
                     if (form.equals("graph")) {
-                        GraphParser gp = getGraphParser();
-                        if (gp == null) {
-                            throw new InvalidStructureException("No graph parser available", parser);
```

Comment:
@orangejenny Means that if you try to install an app w/ graphing on J2ME it won't fail immediately, just if you try to open something that uses the graphing. Hope that is a good failure mode...